src/screens/EditCommunity/EditCommunity.tsx

```diff
@@ -0,0 +1,398 @@
+import * as React from 'react';
+import { useCallback, useEffect, useState } from 'react';
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  Image,
+  TextInput,
+  ScrollView,
+  Pressable,
+  FlatList,
+  Alert,
+} from 'react-native';
+import { SvgXml } from 'react-native-svg';
+import {
+  arrowOutlined,
+  closeIcon,
+  plusIcon,
+  privateIcon,
+  publicIcon,
+} from '../../svg/svg-xml-list';
+import { useStyles } from './styles';
+import ChooseCategoryModal from '../../components/ChooseCategoryModal';
+import { RadioButton } from 'react-native-radio-buttons-group';
+import AddMembersModal from '../../components/AddMembersModal';
+import type { UserInterface } from 'src/types/user.interface';
+import useAuth from '../../hooks/useAuth';
+import { useTheme } from 'react-native-paper';
+import type { MyMD3Theme } from '../../providers/amity-ui-kit-provider';
+import * as ImagePicker from 'expo-image-picker';
+import { uploadImageFile } from '../../providers/file-provider';
+import { getAvatarURL } from '../../util/apiUtil';
+import { updateCommunity } from '../../providers/Social/communities-sdk';
+
+const EditCommunity = ({ navigation, route }) => {
+  const styles = useStyles();
+  const theme = useTheme() as MyMD3Theme;
+  const {
+    communityData: { data },
+  }: { communityData: { data: Amity.RawCommunity } } = route.params;
+  const { apiRegion } = useAuth();
+  const [image, setImage] = useState<string>('');
+  const [communityName, setCommunityName] = useState<string>(data.displayName);
+  const [categoryName, setCategoryName] = useState<string>('');
+  const [categoryId, setCategoryId] = useState<string>(data.categoryIds[0]);
+  const [aboutText, setAboutText] = useState(data.description);
+  const [categoryModal, setCategoryModal] = useState<boolean>(false);
+  const [addMembersModal, setAddMembersModal] = useState<boolean>(false);
+  const [isPublic, setisPublic] = useState(data.isPublic);
+  const [selectedUserList, setSelectedUserList] = useState<UserInterface[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [imageFileId, setImageFileId] = useState<string>(data.avatarFileId);
+
+  const MAX_COMMUNITY_NAME_LENGTH = 30;
+  const MAX_ABOUT_TEXT_LENGTH = 180;
+
+  const isFormValid = (
```

Comment:
it would be better to validate form by this lib https://react-hook-form.com/

Reply:
Thanks. updated krub :)